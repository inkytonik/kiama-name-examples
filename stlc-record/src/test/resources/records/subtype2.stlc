fun(r : {y : num}) { r } {x = 1}

