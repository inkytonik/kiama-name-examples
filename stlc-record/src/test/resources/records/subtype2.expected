src/test/resources/records/subtype2.stlc:1:1:info: fun (r : { ...: unknown
fun(r : {y : num}) { r } {x = 1}
^
src/test/resources/records/subtype2.stlc:1:1:info: fun (r : { ...: {y : num} -> {y : num}
fun(r : {y : num}) { r } {x = 1}
^
src/test/resources/records/subtype2.stlc:1:5:info: r def: argument {y : num}
fun(r : {y : num}) { r } {x = 1}
    ^
src/test/resources/records/subtype2.stlc:1:10:info: y def: field in type num
fun(r : {y : num}) { r } {x = 1}
         ^
src/test/resources/records/subtype2.stlc:1:22:info: r: {y : num}
fun(r : {y : num}) { r } {x = 1}
                     ^
src/test/resources/records/subtype2.stlc:1:26:info: {x = 1}: {x : num}
fun(r : {y : num}) { r } {x = 1}
                         ^
src/test/resources/records/subtype2.stlc:1:27:info: x def: field in value num
fun(r : {y : num}) { r } {x = 1}
                          ^
src/test/resources/records/subtype2.stlc:1:31:info: 1: num
fun(r : {y : num}) { r } {x = 1}
                              ^
unknown
